### PR TITLE
The grand language renaming bonanza

### PR DESCRIPTION
**Pull request by arfon**
Thanks to the stellar work of @chrismwendt and @TwP it's now possible to rename languages on GitHub 🎉 

This is a work-in-progress PR to rename languages that we know are incorrect.

/ cc @pchaigno @larsbrinkhoff @Alhadis as they all seem to have better memories than I :grin:

### CODE REVIEW

**Comment by arfon** [comment]
@larsbrinkhoff in [this comment](https://github.com/github/linguist/issues/2382#issuecomment-101216377) you suggest renaming `Bison` -> `Yacc` - this that still your recommendation?

**Comment by pchaigno** [comment]
Don't we need to wait for the end of the reindexing?

**Comment by arfon** [comment]
> Don't we need to wait for the end of the reindexing?

It's apparently done.

**Comment by pchaigno** [comment]
Wow! That was fast :zap:

**Comment by arfon** [comment]
> Wow! That was fast ⚡️

Yep. So it turns out a there _is_ a little more work that needs doing on GitHub's internal systems before we can merge this PR but in terms of the major effort extracting the language name from the indexes is complete.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -4786,6 +4747,44 @@ reStructuredText:
   codemirror_mode: rst
   codemirror_mime_type: text/x-rst
   language_id: 419
+roff:
```
Comment (Alhadis) [inline lib/linguist/languages.yml]: Uppercase `Roff`, not `roff`. =)

Comment (arfon) [inline lib/linguist/languages.yml]: Can't tell if you're trolling me now @Alhadis 😁 

I thought `roff` was the consensus here: https://github.com/github/linguist/issues/2382#issuecomment-196793686

Comment (Alhadis) [inline lib/linguist/languages.yml]: Yeah, that probably should've been capitalised too. 😁  My bad.

**Comment by Alhadis** [comment]
[Interesting topic](http://stackoverflow.com/questions/4398312/vimscript-or-viml#31422313) concerning Vim script. It'd appear that we're to blame for the **VimL** construction, heh.

> The relatively new name “VimL” (“Vim Language”) has been gaining in popularity in rough correlation with the growth of the code-sharing site GitHub.

**Comment by arfon** [comment]
> Interesting topic concerning Vim script. It'd appear that we're to blame for the VimL construction, heh.

Yeah I read that. Not entirely convinced we're making things any better here with this change `¯\_(ツ)_/¯`

**Comment by Alhadis** [comment]
Me neither, but it's been my experience that any language without an obvious, definitive name becomes a pain-in-the-khyber. 🙁

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -1417,6 +1407,16 @@ Gettext Catalog:
   tm_scope: source.po
   ace_mode: text
   language_id: 129
+Gherkin:
+  type: programming
+  extensions:
+  - ".feature"
+  tm_scope: text.gherkin.feature
+  aliases:
+  - cucumber
```
Comment (larsbrinkhoff) [inline lib/linguist/languages.yml]: Yes.  Bison is just the GNU project implementation of the Yacc language.  But I see now that there already is a `Yacc` in linguist.  Maybe it was added later.  And that Bison is grouped under Yacc.  So I don't think there's anything to do.

**Comment by larsbrinkhoff** [comment]
@arfon Bison is just the GNU project implementation of the Yacc language. But I see now that there already is a Yacc in linguist. Maybe it was added later. And that Bison is grouped under Yacc. So I don't think there's anything to do.

**Comment by pbregener** [comment]
What's the status here @arfon? Seems like some name changes are more controversial than others. Could it make sense to merge the uncontroversial part very soon and discuss the remaining parts at length elsewhere?

**Comment by Alhadis** [comment]
@pbregener Can you define "controversial"?

**Comment by pbregener** [comment]
@Alhadis Well, "controversial" is probably not the right word. It just seems like there is some confusion in this thread, e.g. concerning the renaming of the "Vim Script Language". On the other hand, probably everyone agrees that it is time to rename "FORTRAN" to "Fortran".

**Comment by arfon** [comment]
> What's the status here @arfon? Seems like some name changes are more controversial than others. Could it make sense to merge the uncontroversial part very soon and discuss the remaining parts at length elsewhere?

We're still waiting on some internal refactors of GitHub's codebase before this can be merged.

**Comment by pbregener** [comment]
Thanks @arfon. Is there any ETA on this?

**Comment by arfon** [comment]
> Thanks @arfon. Is there any ETA on this?

It's on our engineering roadmap for the search team (I'm not in this group). I don't have much more information that that sorry.

/ @TwP or @chrismwendt incase they have more information.

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -3568,6 +3520,44 @@ RobotFramework:
   tm_scope: text.robot
   ace_mode: text
   language_id: 324
+Roff:
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: According to #2382, this should be `roff`, without upper-case.

Comment (Alhadis) [inline lib/linguist/languages.yml]: That's it, I've edited my comment so this won't be mistaken a third time.

Comment (pchaigno) [inline lib/linguist/languages.yml]: Oops! Never mind :innocent:

**Pull request merged**